src/frontend/src/utils/group_by.rs

```diff
@@ -0,0 +1,34 @@
+use crate::expr::ExprImpl;
+
+#[derive(Debug, Clone)]
+pub enum GroupBy {
+    GroupKey(Vec<ExprImpl>),
+    GroupingSets(Vec<Vec<ExprImpl>>),
+}
+
+impl GroupBy {
+    pub fn is_empty(&self) -> bool {
+        match self {
+            GroupBy::GroupKey(group_key) => group_key.is_empty(),
+            GroupBy::GroupingSets(grouping_sets) => grouping_sets.is_empty(),
+        }
+    }
+
+    pub fn iter(&self) -> Box<dyn Iterator<Item = &ExprImpl> + '_> {
```

Comment:
nit: we can use `#[auto_enum(Iterator)]` for this.

Reply:
That's cool. Let me refactor by `#[auto_enum(Iterator)]`.